Bug Report: invalid memory address or nil pointer dereference

## Summary of Bug 
Multiple runtime errors

## Steps to Reproduce

1. start tendermint node
2. start kvstore in cosmos/example

3. open url: http://tendermint-node-ip:26657/abci_query?path="custom"

 ```panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x17e516f]

goroutine 37 [running]:
github.com/cosmos/cosmos-sdk/baseapp.handleQueryCustom(0xc00024fd10, 0xc000934060, 0x2, 0x2, 0x0, 0x0, 0x0, 0xc000932240, 0x7, 0x0, ...)
	/workspace/cosmos-sdk/baseapp/baseapp.go:399 +0x2cf
github.com/cosmos/cosmos-sdk/baseapp.(*BaseApp).Query(0xc00024fd10, 0x0, 0x0, 0x0, 0xc000932240, 0x7, 0x0, 0x1, 0x0, 0x0, ...)
	/workspace/cosmos-sdk/baseapp/baseapp.go:311 +0x327
github.com/tendermint/tendermint/abci/server.(*SocketServer).handleRequest(0xc0000b2c80, 0xc00019b080, 0xc000196600)
	/workspace/gocode/pkg/mod/github.com/tendermint/tendermint@v0.25.0/abci/server/socket_server.go:190 +0xd7b
github.com/tendermint/tendermint/abci/server.(*SocketServer).handleRequests(0xc0000b2c80, 0xc0001965a0, 0x1beacc0, 0xc0001b60b0, 0xc000196600)
	/workspace/gocode/pkg/mod/github.com/tendermint/tendermint@v0.25.0/abci/server/socket_server.go:163 +0xe5
created by github.com/tendermint/tendermint/abci/server.(*SocketServer).acceptConnectionsRoutine
	/workspace/gocode/pkg/mod/github.com/tendermint/tendermint@v0.25.0/abci/server/socket_server.go:119 +0x269

Process finished with exit code 2
```
4. open url: http://tendermint-node-ip:26657/abci_query?path="custom/name"

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x17e516f]

goroutine 22 [running]:
github.com/cosmos/cosmos-sdk/baseapp.handleQueryCustom(0xc00024ee10, 0xc0010e6760, 0x2, 0x2, 0x0, 0x0, 0x0, 0xc0001ef970, 0xb, 0x0, ...)
	/workspace/cosmos-sdk/baseapp/baseapp.go:399 +0x2cf
github.com/cosmos/cosmos-sdk/baseapp.(*BaseApp).Query(0xc00024ee10, 0x0, 0x0, 0x0, 0xc0001ef970, 0xb, 0x0, 0x1, 0x0, 0x0, ...)
	/workspace/cosmos-sdk/baseapp/baseapp.go:311 +0x327
github.com/tendermint/tendermint/abci/server.(*SocketServer).handleRequest(0xc0000c28c0, 0xc00108c0c0, 0xc0000b6840)
	/workspace/gocode/pkg/mod/github.com/tendermint/tendermint@v0.25.0/abci/server/socket_server.go:190 +0xd7b
github.com/tendermint/tendermint/abci/server.(*SocketServer).handleRequests(0xc0000c28c0, 0xc0000b67e0, 0x1beacc0, 0xc0001c20c0, 0xc0000b6840)
	/workspace/gocode/pkg/mod/github.com/tendermint/tendermint@v0.25.0/abci/server/socket_server.go:163 +0xe5
created by github.com/tendermint/tendermint/abci/server.(*SocketServer).acceptConnectionsRoutine
	/workspace/gocode/pkg/mod/github.com/tendermint/tendermint@v0.25.0/abci/server/socket_server.go:119 +0x269

Process finished with exit code 2
```

--- a/baseapp/baseapp.go
+++ b/baseapp/baseapp.go
@@ -384,12 +384,12 @@ func handleQueryP2P(app *BaseApp, path []string, req abci.RequestQuery) (res abc
 func handleQueryCustom(app *BaseApp, path []string, req abci.RequestQuery) (res abci.ResponseQuery) {
 	// path[0] should be "custom" because "/custom" prefix is required for keeper queries.
 	// the queryRouter routes using path[1]. For example, in the path "custom/gov/proposal", queryRouter routes using "gov"
-	if path[1] == "" {
-		sdk.ErrUnknownRequest("No route for custom query specified").QueryResult()
+	if len(path) < 2 || path[1] == "" {
+		return sdk.ErrUnknownRequest("No route for custom query specified").QueryResult()
 	}
 	querier := app.queryRouter.Route(path[1])
 	if querier == nil {
-		sdk.ErrUnknownRequest(fmt.Sprintf("no custom querier found for route %s", path[1])).QueryResult()
+		return sdk.ErrUnknownRequest(fmt.Sprintf("no custom querier found for route %s", path[1])).QueryResult()
 	}
 
 	ctx := sdk.NewContext(app.cms.CacheMultiStore(), app.checkState.ctx.BlockHeader(), true, app.Logger).

